config/filter.d/apache-noindex.conf

```diff
@@ -0,0 +1,19 @@
+# Fail2Ban filter to block unintended requests for directory browsing
+#
+# Directory browsing is ontrolled by Indexes Option in apache2. The option
+# can by set per directory. This filter helps to block unintended requests.
+# for direcotry browsing.
+
+[INCLUDES]
+
+# overwrite with apache-common.local if _apache_error_client is incorrect.
+before = apache-common.conf
+
+[Definition]
+
+failregex = ^%(_apache_error_client)s ((AH01276: )?.*Cannot serve directory .* directory index forbidden by Options directive$|Directory index forbidden by Options directive: .*$)
```

Comment:
Sorry, but no, this regex is too vulnerable has many catch-alls and therefore the decision-making parts are not anchored... (let alone it'd be slow on parasitic messages).
Something like this maybe?
```diff
- failregex = ^%(_apache_error_client)s ((AH01276: )?.*Cannot serve directory .* directory index forbidden by Options directive$|Directory index forbidden by Options directive: .*$)
+ failregex = ^%(_apache_error_client)s (?:(?:AH01276: )?Cannot serve directory \S+: No matching DirectoryIndex \([^\)]*\) found, and server-generated directory|Directory) index forbidden by Options directive\b
```